### PR TITLE
Timer

### PR DESCRIPTION
**Pull request by asaxena5**
### Summary 
Added basic timer

### Testing 
Runs as normal, and from what I can tell, the timing is synced up.

### Review Focus 
N/A

### CODE REVIEW

File: source/GameMode.cpp [hunk 1]
```diff
@@ -39,14 +39,16 @@ constexpr unsigned int MAX_EVENTS = 3;
...
-/** The Angle in degrees for which a door can be activated*/
-constexpr float DOOR_ACTIVE_ANGLE = 15.0f;
+/** The Angle in radians for which a tap can registers as fixing a breach*/
+constexpr float EPSILON_ANGLE = 0.09f;
+/** The Angle in radians for which a collision occurs*/
+constexpr float DOOR_WIDTH = 0.12f;
+/** The Angle in radians for which a breach donut collision occurs*/
+constexpr float BREACH_WIDTH = 0.2f;
+/** The Angle in radians for which a door can be activated*/
+constexpr float DOOR_ACTIVE_ANGLE = 0.25f;
```
Comment (mt-xing) [inline source/GameMode.cpp:49]: Why'd this get switched back to radians?

Comment (asaxena5) [reply source/GameMode.cpp:49]: shoot that might've been me messing up the merge conflicts I'll fix that

**Comment by mt-xing** [comment]
Something's wrong with the CI job for `sweetspace-compile`. I'm going to kill and re-issue it.

**Comment by mt-xing** [comment]
Alright, so the CI pipeline is just being screwy right now. It compiles fine on my machine; I'm just going to merge this in.

**Pull request merged**